src/WebJobs.Extensions/Extensions/Timers/Listener/TimerListener.cs

```diff
@@ -173,10 +173,14 @@ private void OnTimer(object sender, ElapsedEventArgs e)
                 return;
             }
 
-            // restart the timer with the next schedule occurrence
             await InvokeJobFunction(DateTime.Now, false);
 
-            StartTimer(DateTime.Now);
+            // Restart the timer with the next schedule occurrence, but only 
+            // if Cancel, Stop, and Dispose have not been called.
+            if (!_cancellationTokenSource.IsCancellationRequested)
```

Comment:
Should this check be moved into StartTimer? I see another StartTimer call above that isn't guarded.